src/test/common.ts

```diff
@@ -162,3 +162,45 @@ export function createEventHandler<T, K extends keyof T>(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     return new TestEventHandler(obj[eventName] as any, eventName as string, disposables) as any;
 }
+/**
+ * API common to web & desktop tests, but with different implementations
+ */
+export type CommonApi = {
+    createTemporaryFile(options: { extension: string; contents?: string }): Promise<{ file: Uri } & IDisposable>;
+    startJupyterServer(notebook?: NotebookDocument, useCert?: boolean): Promise<void>;
+    stopJupyterServer?(): Promise<void>;
+    captureScreenShot?(fileNamePrefix: string): Promise<void>;
+    initialize(): Promise<IExtensionTestApi>;
```

Comment:
These are the functions that are different on each platform, and if we had a common function for them, then we won't have to create two 3 seprate files for the tests, now that we have a common layer for these functions, we have just one single file (tests that run on web & desktop)